src/figuresClasses.ts

```diff
@@ -2,10 +1,97 @@
+export enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
+
+export enum Color {
+  Red = 'red',
+  Green = 'green',
+  Blue = 'blue',
+}
+
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
 
-export class Triangle implements Figure {}
+export class Triangle implements Figure {
+  shape: Shape = Shape.Triangle;
 
-export class Circle implements Figure {}
+  constructor(
+    public color: Color,
+    protected sideA: number,
+    protected sideB: number,
+    protected sideC: number,
+  ) {
+    if (sideA <= 0 || sideB <= 0 || sideC <= 0) {
+      throw new Error('Triangle side lengths must all be greater than 0.');
+    }
 
-export class Rectangle implements Figure {}
+    const longestSide = Math.max(sideA, sideB, sideC);
+    const perimeter = sideA + sideB + sideC;
+
+    if (longestSide >= perimeter - longestSide) {
+      throw new Error(
+        `The longest side (${longestSide}) must be shorter than the sum of the other two sides (${sideA + sideB + sideC - longestSide}).`,
+      );
+    }
+  }
+
+  getArea(): number {
```

Comment:
[CODE STYLE] - The `getArea` function shouldn't accept any parameters as all necessary values are already available as object fields. Since you're using `this` to access the object's fields inside the method, no parameters are needed.